library/lua/dfhack.lua

```diff
@@ -451,6 +526,9 @@ end
 
 -- Removes spaces (i.e. everything that matches '%s') from the start and end of
 -- a string. Spaces between non-space characters are left untouched.
+---@nodiscard
+---@param self string
+---@return self
```

Comment:
for these `string` functions, it doesn't return `self`, it returns a new `string`